spec/post_spec.rb

```diff
@@ -1,34 +1,41 @@
 require 'rails_helper'
 
-describe Post, type: :model do
-  context 'test posts' do
-    before :each do
-      @user = User.create(name: 'william', photo: 'https://github.com/kidde60', bio: 'web developer')
-      @post = Post.create(author: @user, title: 'Title', text: 'My sample text')
+RSpec.describe Post, type: :model do
+  let(:user) { User.create(name: 'Test User', photo: 'test@example.com', bio: 'password', posts_counter: 0) }
+  let(:post) { Post.create(title: 'Test Post', text: 'Test content', author: user) }
+
+  context '#post_counter' do
+    it 'increments the author post counter' do
+      expect { post.posts_counter }.to change { user.reload.posts_counter }.by(1)
     end
+  end
+
+  context '#five_most_recent_comments' do
+    it 'returns the 5 most recent comments' do
+      comments = post.five_most_recent_comments
+      expect(comments).to eq(post.comments.order(created_at: :desc).limit(5))
+    end
+  end
 
-    it 'should not have an empty title' do
-      expect(@post.title).to eql 'Title'
+  context 'validations' do
+    it 'requires a title' do
+      post.title = ''
+      expect(post).to_not be_valid
     end
 
-    it 'should have 0 comments at default' do
-      expect(@post.comments_counter).to eq 0
+    it 'requires title to be no more than 250 characters' do
+      post.title = 'a' * 251
+      expect(post).to_not be_valid
     end
 
```

Comment:
- [OPTIONAL] Since I'm not able to inline comment on your license I just do it here but I noticed that the link provided is not working properly, Kindly fix this as it was not flagged up by the previous review and is mandatory requirement for this project :wink: 
![Screenshot from 2023-02-25 21-57-00](https://user-images.githubusercontent.com/88894525/221377253-c81b98bb-e0ed-4c4c-925f-13d2d398f061.png)

_note: I just noticed that you  are using a broken url in `README` file the logo image is not working , I suggest you to fix that and use your own logo for your Readme to look more professional :wink:_
![Screenshot from 2023-02-25 22-01-08](https://user-images.githubusercontent.com/88894525/221377411-f73bed2b-55cd-4f4c-8c67-874c6572cc0f.png)